lua/header.lua

```diff
@@ -160,7 +167,40 @@ local function add_license_header(opts)
         local commented_headers = comment_headers(license_table, comments)
         vim.api.nvim_buf_set_lines(buffer, 0, 0, false, commented_headers)
     else
-        print("Unsupported file type:", file_extension)
+        vim.notify_once("unsupported file type for adding header", vim.log.levels.ERROR)
+    end
+end
+
+function header.update_date_modified()
+    local buffer = vim.api.nvim_get_current_buf()
+    local file_extension = vim.fn.expand("%:e")
+    local comments = filetype_table[file_extension]() -- Retrieve the comment style for the current file type
+    local lines = vim.api.nvim_buf_get_lines(buffer, 0, 8, false)
+
+    -- First, determine if there is a header by looking for the "Date created" line
+    local has_header = false
+    for _, line in ipairs(lines) do
+        if line:find(header.constants.date_created) then
+            has_header = true
+            break
+        end
+    end
+
+    -- If a header is found, then proceed to update the "Date modified" line
+    if has_header then
+        local header_end = find_header_end(lines, comments)
+        local modified_date = os.date(header.config.date_modified_fmt)
+
+        for i, line in ipairs(lines) do
+            if line:find(header.constants.date_modified) then
+                -- Update the "Date modified" line with the new date and the comment syntax
+                lines[i] = comments.comment .. " " .. header.constants.date_modified .. " " .. modified_date
+                break
+            end
+        end
+
+        -- Replace only the header lines in the buffer
+        vim.api.nvim_buf_set_lines(buffer, 0, header_end, false, lines)
```

Comment:
```suggestion
        vim.api.nvim_buf_set_lines(buffer, 0, header_end + 1, false, lines)
```
without the addition of `+1` the update header method constantly introduces newlines appended to the source file. this is why I always clear the headers first, then recreate it.